### PR TITLE
Make debugging BDDs using regular expressions easier

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Make debugging BDDs using regular expressions easier

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by heshanpadmasiri** [comment]
@jclark this will make running `nballerina.regex.jar` with `--balTypes` to add bdd structure to each type definition as a comment. Example:
```
// ((((Tcn)*ke)*)*S)*v
type T3 T6 | T2; // r0?1:(r3?1:(r5?1:(r6?1:0:0):0):0):0
type T6 T11 | T5; // r0?1:(r3?1:(r5?1:0:0):0):0
type T11 T17 | T10; // r0?1:(r3?1:0:0):0
type T17 [T12, T16]; // r0?1:0:0
type T12 "T";
type T16 [T13, T15]; // r1?1:0:0
type T13 "c";
type T15 [T14, T11]; // r2?1:0:0
type T14 "n";
type T10 [T7, T9]; // r3?1:0:0
type T7 "k";
type T9 [T8, T6]; // r4?1:0:0
type T8 "e";
type T5 [T4, T3]; // r5?1:0:0
type T4 "S";
type T2 [T1, T0]; // r6?1:0:0
type T1 "v";
type T0 ();
// (((((CvGb)*Tcn)*ke)*)*S)*v
type T21 T24 | T20; // r7?1:(r11?1:(r14?1:(r16?1:(r17?1:0:0):0):0):0):0
type T24 T29 | T23; // r7?1:(r11?1:(r14?1:(r16?1:0:0):0):0):0
type T29 T36 | T28; // r7?1:(r11?1:(r14?1:0:0):0):0
type T36 T44 | T35; // r7?1:(r11?1:0:0):0
type T44 [T37, T43]; // r7?1:0:0
type T37 "C";
type T43 [T38, T42]; // r8?1:0:0
type T38 "v";
type T42 [T39, T41]; // r9?1:0:0
type T39 "G";
type T41 [T40, T36]; // r10?1:0:0
type T40 "b";
type T35 [T30, T34]; // r11?1:0:0
type T30 "T";
type T34 [T31, T33]; // r12?1:0:0
type T31 "c";
type T33 [T32, T29]; // r13?1:0:0
type T32 "n";
type T28 [T25, T27]; // r14?1:0:0
type T25 "k";
type T27 [T26, T24]; // r15?1:0:0
type T26 "e";
type T23 [T22, T21]; // r16?1:0:0
type T22 "S";
type T20 [T19, T18]; // r17?1:0:0
type T19 "v";
type T18 ();

```

File: compiler/modules/types/bdd.bal [hunk 1]
```diff
@@ -227,3 +227,15 @@ isolated function bddToString(Bdd b, boolean inner = false) returns string {
         return str;
     }
 }
+
+// This is for debug purposes. Used by regex to add comments for type definitions
+public isolated function semTypeToBddString(SemType ty) returns string? {
+    if ty !is ComplexSemType {
+        return ();
+    } 
+    var currentSubtype = ty.subtypeDataList[0];
```
Comment (jclark) [inline compiler/modules/types/bdd.bal:236]: This isn't in general going to be right.

I think it's good to have public function in core that gives a human understandable version of the semType, but it needs to be a bit more sophisticated than this, if we are going to merge it in.

**Review by jclark — COMMENTED** [review]
This is some way from something I would want to merge in.

**Comment by heshanpadmasiri** [comment]
Closing this because the original problem we were trying to figure out (#1076) no longer needs this.

**Pull request closed without merging**